tests/rptest/services/redpanda.py

```diff
@@ -1586,6 +1590,19 @@ def stop(self, **kwargs):
                         lambda n: self.stop_node(n, **kwargs),
                         parallel=True)
 
+        if len(self.failure_injectors) > 0:
```

Comment:
I don't think this is the best place to put this, since there's no guarantee the intention upon calling `stop()` is to destruct test state. At least, this is used in some test bodies:

```grep -r "redpanda\.stop(" tests/rptest | wc -l
4
```

Maybe we should put this in services/cluster.py, where we handle some other post-test tasks?